### PR TITLE
Fixes #14878: Add a contribution guide to the Rudder repo

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/14878

### CODE REVIEW

File: CONTRIBUTING.adoc [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+* Documentation
+* Features ideas
+* New features
+
+If you want to contribute, the best way to start is to come and chat with us
+on https://chat.rudder.io[chat.rudder.io]. You can also communicate by email
+using the  https://www.rudder-project.org/mailman/listinfo/rudder-dev[rudder-dev]
+mailing list.
+
+NOTE: Contribution to Rudder requires a https://www.rudder.io/en/expand/contribute/#panel-2422-8-0-1[CLA/CCLA].
```
Comment (fanf) [inline CONTRIBUTING.adoc]: I would say: 
Large contribution on Rudder may requires...
Small / trivial contrib don't require the CCLA, and we don't want to frighten someone who just want to correct a typo in somewhere.

File: CONTRIBUTING.adoc [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+== Development workflow
```
Comment (fanf) [inline CONTRIBUTING.adoc]: Before that, I would had: "documentation workflow" with "you can just click edit on the doc". To start with the easy parts :)

File: CONTRIBUTING.adoc [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+We maintain several Rudder major versions in separate branches.
+
+* `branches/rudder/5.0` is Rudder 5.0, from which 5.0 minor release will be tagged
+* `master` is the next Rudder version, not branched yet
+
+NOTE: Some tooling repositories only have a `master` branch.
+
+All changes need to be done in the oldest relevant branch (i.e.
```
Comment (fanf) [inline CONTRIBUTING.adoc]: s/changes need to/bug correction should/
And perhaps something explaining that we can assist them if they fear to deal with that oldest-branch rule.

File: CONTRIBUTING.adoc [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+```
+wget https://repository.rudder.io/tools/rudder-dev
+```
+
+It is a shell script you can put somewhere is your PATH. You can then update it with
+
+```
+rudder-dev update
+```
+
```
Comment (fanf) [inline CONTRIBUTING.adoc]: You should tell here that there's a configuration file to update with github token.

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**